mesonbuild/dependencies/dev.py

```diff
@@ -568,6 +568,18 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
 
         m = self.env.machines[self.for_machine]
 
+        if m.is_android():
+            if 'version' in kwargs:
+                mlog.error('Can\'t set version filter for jni on Android')
+                self.is_found = False
+                return
```

Comment:
I think these need to be `if kwargs.get('version'):` because they may get set to `None` or `[]` in some paths.

I guess the version can't be found without actually compiling something here, since jni.h doesn't have any versions in it.

I think for version I'd prefer a warning rather than an exception, since this prevents someone from having code that runs on Android and !Android without doing extra work. So instead of users having to deal with that, just doing the:
```python
mlog.debug("Android does not provide a mechanism to test the version of JNI, version restrictions are ignored")
```
Is a better solution. Also just debug for the debug log, because it's not an actionable item.

Reply:
I mean jni.h kinda has version constants (the one you pass to AttachThread to specify what JNI interface version you want), its just that this doesn't accurately reflect the java version the version field currently uses (By just querying `javac --version`).

This obv. won't work for Android, given that you first won't know the java version of the host device and second that they don't use the JVM altogether but instead their ART which just defines the same jni interface)

You could in theory map the different java versions to the latest jni version constant it introduced and check if it is available, but I'm not sure if it is worth it.

Reply:
Yeah, that all makes sense (and I expected this to be an ART issue), so I think I still like just ignoring it and hoping that you've done the work ahead of time to make sure everything works out is really the best we can do. Especially since someone could really want to target both android and something else in the same code base.